vllm/v1/worker/gpu_model_runner.py

```diff
@@ -2998,6 +3000,36 @@ def execute_model(
             # Mark KV scales as calculated after the first forward pass
             self.calculate_kv_scales = False
 
+        if (
```

Comment:
Please refactor this out of `execute_model` and into a helper function. We want to keep critical code pathways brief and readable.